src/App.tsx

```diff
@@ -63,13 +86,40 @@ export default function App(): JSX.Element {
   }, []);
 
   useEffect(() => {
-    setBeaconApiUrl(getNetworkUrl(network));
+    setBeaconApiUrl(getNetworkUrl(network).beaconApiUrl);
+    setElRpcUrl(getNetworkUrl(network).elRpcUrl);
   }, [network]);
 
-  // Check if local snapshot is available
   useEffect(() => {
-    setLocalAvailable(hasSnapshot());
-  }, [reqStatusInit.result]);
+    const client = reqStatusInit.result;
+    if (client && head && address && elRpcUrl) {
+      try {
+        if (!web3.current) web3.current = new Web3(elRpcUrl);
+      } catch (e) {
+        setAccountReqStatus({result: accountReqStatus.result, error: e as Error});
+        return;
+      }
+      const blockHash = toHexString(ssz.phase0.BeaconBlockHeader.hashTreeRoot(head));
+      client.api.beacon.getBlockV2(blockHash).then((data) => {
+        const {data: block} = data as unknown as {data: bellatrix.SignedBeaconBlock};
+        const executionPayload = block.message.body.executionPayload;
+        // If the merge not complete, executionPayload would not exists
+        if (!executionPayload) {
+          setAccountReqStatus({result: accountReqStatus.result, loading: `Waiting for an execution payload`});
+          return;
+        } else {
+          setAccountReqStatus({result: accountReqStatus.result, loading: `Fetching status from ${elRpcUrl}`});
+          fetchAndVerifyAddress({web3: web3.current, executionPayload, address})
+            .then((verifiedAccount) => {
+              setAccountReqStatus({result: verifiedAccount});
+            })
+            .catch((e) => {
+              setAccountReqStatus({result: accountReqStatus.result, error: e});
```

Comment:
Recommended pattern in React async useEffect is to just declare and async function in the useEffect body and call it with a .catch() then you can use await